renderdoc/driver/d3d12/d3d12_device_wrap.cpp

```diff
@@ -582,7 +582,9 @@ void WrappedID3D12Device::CreateShaderResourceView(ID3D12Resource *pResource,
                                                    D3D12_CPU_DESCRIPTOR_HANDLE DestDescriptor)
 {
   GetWrapped(DestDescriptor)->Init(pResource, pDesc);
-  return m_pDevice->CreateShaderResourceView(Unwrap(pResource), pDesc, Unwrap(DestDescriptor));
+  if(!pResource)
+    return;
```

Comment:
I guess this is coming from `D3D12Descriptor::Create` where we create a SRV with a NULL resource for undefined descriptors?

I think we'll need those to be well-defined empty descriptors rather than invalid uninitialised ones. I think the proper fix here would be to determine the heap type and create the appropriate NULL resource descriptor (since it doesn't matter if it's SRV or what, just that it's valid).

I believe doing this early return will prevent any application-created descriptors with a NULL resource from initializing, which could lead to crashes when they're used in rendering.